homeassistant/components/withings/sensor.py

```diff
@@ -0,0 +1,454 @@
+"""Sensors flow for Withings."""
+import typing as types
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util import slugify
+
+from . import const
+from .common import _LOGGER, create_withings_data_manger, WithingsDataManager
+
+# There's only 3 calls (per profile) made to the withings api every 5
+# minutes (see throttle values). This component wouldn't benefit
+# much from parallel updates.
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: types.Callable[[types.List[Entity], bool], None],
+):
+    """Set up the sensor config entry."""
+    data_manager = create_withings_data_manger(hass, entry)
```

Comment:
If we would store the data manager instance in hass.data we wouldn't need to create it again here after forwarding the config entry.